lib/operators/operatorDateCompare.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const moment = require('moment');
+
+function getAnswer(questionDotPointer, answers) {
+    let value = false;
+    if (!questionDotPointer) {
+        return value;
+    }
+    const questionId = questionDotPointer.split('.')[3];
+    Object.keys(answers).forEach(page => {
+        if (questionId in answers[page]) {
+            value = answers[page][questionId];
+        }
+    });
+    return value;
+}
+
+function dateCompare(rule, data) {
+    const inputDate = getAnswer(rule[1], data.answers); // LHS.
+    // direction of comparison. e.g. '>' means: is `inputDate` more than `amount` of `units`  different from `compareDate`.
+    // i.e. is '2005-02-01T00:00:00.000Z' more than -12 years different from (i.e. 12 years before) '2021-02-01T00:00:00.000Z' - this evaluates to `true`.
```

Comment:
Does this matter given you're using `Math.abs()`? Are we are gaining anything by including the negative numbers convention.

Reply:
this is to enable absolute comparisons. e.g.: 
* "less than" 12 years ago (-12)
* "less than" 12 years from now (12)

the "les than (<) is not an indication of which way "in time" the comparison is